### PR TITLE
Generate linting reports and fail pipeline in same step

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

Improvement to CI reporting of linting issues to have the report generation and failing of the pipeline due to linting errors happen in the same step. This removes confusion where the reviewer of the failed pipeline has to figure out DevOps artifact semantics, aka where to download the linting-reports.zip, and presents the errors in an obvious location. The `linting-reports.zip` will still be generated but the Checkstyle, RevApi, and Spotbugs errors will be shown in the step that has failed the pipeline as well, making them more obvious.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: eng/pipelines/scripts/Get-Linting-Reports.ps1 [hunk 1]
```diff
@@ -154,3 +154,7 @@ if (-not (Test-Path "$StagingDirectory/troubleshooting")) {
 
 Compress-Archive -Path $OutputDirectory -DestinationPath "$StagingDirectory/troubleshooting/linting-report.zip"
 Write-Host "##vso[task.setvariable variable=HAS_TROUBLESHOOTING]true"
+
+# Fail the build if any of the reports contain errors.
+Write-Host "Linting errors have been found, download the 'linting-report.zip' artifact to see what was found."
```
Comment (joshfree) [inline eng/pipelines/scripts/Get-Linting-Reports.ps1:159]: Can the error message have the full path to the file to make it easier to copy/paste?

**Review by joshfree — APPROVED** [review]

File: eng/pipelines/scripts/Get-Linting-Reports.ps1 (hunk 1 above)
Comment (alzimmermsft) [reply eng/pipelines/scripts/Get-Linting-Reports.ps1:159]: This gets published as a DevOps artifact, I don't think there's a way to create a link to it at this point.

Comment (benbp) [reply eng/pipelines/scripts/Get-Linting-Reports.ps1:159]: The actual download URLs are too dynamic to hardlink to, but you could link to the artifacts, though I don't see a way to link to a specific path/directory within the artifacts.

```
https://dev.azure.com/azure-sdk/$(System.TeamProject)/_build/results?buildId=$(Build.BuildId)&view=artifacts&pathAsName=false&type=publishedArtifacts
```

Comment (benbp) [inline eng/pipelines/scripts/Get-Linting-Reports.ps1:159]:
```suggestion
Write-Warning "Linting errors have been found, download the 'linting-report.zip' artifact to see what was found."
```

**Review by benbp — APPROVED** [review]

**Pull request merged**